apps/sms/test/marionette/notification_test.js

```diff
@@ -28,6 +29,29 @@ marionette('Message notification tests', function() {
     }), message);
   }
 
+  function assertMessagesIsInCorrectStateFromMessage(smsMessage) {
+    // Verify that we entered the right thread
+    var message = messagesApp.Conversation.message;
+    assert.equal(
+      messagesApp.Conversation.getMessageContent(message).text(),
+      smsMessage.body
+    );
+
+    assert.isFalse(messagesApp.Composer.sendButton.enabled());
+
+    // Validate that header is correctly set even that thread list hasn't
+    // been loaded yet.
+    assert.equal(
+      messagesApp.Conversation.headerTitle.text(), smsMessage.sender
+    );
+
+    // When we open thread from notification, message input should not be
+    // focused.
+    assertIsNotFocused(
+      messagesApp.Composer.messageInput, 'Message input should not be focused'
+    );
+  }
```

Comment:
I changed this in a previous version of the patch, and then I left it even than I don't need it. Tell me if you'd rather want that I revert the change.